src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.HelperFunctions.cs

```diff
@@ -84,15 +88,28 @@ private ExecutionPayload CreateParentBlockRequestOnHead(IBlockTree blockTree)
             };
         }
 
-        private static ExecutionPayload CreateBlockRequest(ExecutionPayload parent, Address miner, IList<Withdrawal>? withdrawals = null,
+        private static ExecutionPayload CreateBlockRequest(IReleaseSpec spec, IWorldState state, ExecutionPayload parent, Address miner, IList<Withdrawal>? withdrawals = null,
                 ulong? blobGasUsed = null, ulong? excessBlobGas = null, Transaction[]? transactions = null, Keccak? parentBeaconBlockRoot = null)
-            => CreateBlockRequestInternal<ExecutionPayload>(parent, miner, withdrawals, blobGasUsed, excessBlobGas, transactions: transactions, parentBeaconBlockRoot: parentBeaconBlockRoot);
+            => CreateBlockRequestInternal<ExecutionPayload>(spec, state, parent, miner, withdrawals, blobGasUsed, excessBlobGas, transactions: transactions, parentBeaconBlockRoot: parentBeaconBlockRoot);
 
-        private static ExecutionPayloadV3 CreateBlockRequestV3(ExecutionPayload parent, Address miner, IList<Withdrawal>? withdrawals = null,
+        private static ExecutionPayloadV3 CreateBlockRequestV3(IReleaseSpec spec, IWorldState state, ExecutionPayload parent, Address miner, IList<Withdrawal>? withdrawals = null,
                 ulong? blobGasUsed = null, ulong? excessBlobGas = null, Transaction[]? transactions = null, Keccak? parentBeaconBlockRoot = null)
-            => CreateBlockRequestInternal<ExecutionPayloadV3>(parent, miner, withdrawals, blobGasUsed, excessBlobGas, transactions: transactions, parentBeaconBlockRoot: parentBeaconBlockRoot);
+        {
+            var blockRequestV3 = CreateBlockRequestInternal<ExecutionPayloadV3>(spec, state, parent, miner, withdrawals, blobGasUsed, excessBlobGas, transactions: transactions, parentBeaconBlockRoot: parentBeaconBlockRoot);
+            blockRequestV3.TryGetBlock(out Block? block);
+            _beaconBlockRootHandler.InitStatefulPrecompiles(block!, spec, state);
+
+            state.Commit(spec);
+            state.CommitTree(blockRequestV3.BlockNumber);
+
+            state.RecalculateStateRoot();
+            blockRequestV3.StateRoot = state.StateRoot;
+            TryCalculateHash(blockRequestV3, out Keccak? hash);
```

Comment:
We recalculate it below `CreateBlockRequestInternal<T>` btw. What is interesting is that we do not execute transactions that may affect the state. Maybe state hash root was already calculated before( _beaconBlockRootHandler can be placed there if so)?

Reply:
@Demuirgos ?

Reply:
I am not sure tbh, I'll have to take a look into it